utils/build_env.py

```diff
@@ -57,8 +56,9 @@ def _env(*arg, **kwargs):
 
 print('build %s (settings from: %s)' % (brand_env, os.environ['SEARX_SETTINGS_PATH']))
 sys.path.insert(0, repo_root)
-from searx import settings
```

Comment:
This is the fragile part in the old implementation, read the comments in the lines before.  The problem is, that the settings.yml is imported as soon as you import any searx modules.  If the buildenv should work, you first need to remove any environment (see code above) before importing searx, otherwise the environment will overwrite the values you have modified in the settings.yml.

I fixed it in my branch https://github.com/return42/searxng/commit/6a2d4915a5a8698083d9b21cf6d89c95d2b3caf7